docs/connector-development/config-based/understanding-the-yaml-file/yaml-overview.md

```diff
@@ -4,147 +4,174 @@ The low-code framework involves editing a boilerplate [YAML file](../low-code-cd
-3. [Schema](../../cdk-python/schemas.md): Describes the data to sync
-4. [Data retriever](#data-retriever): Describes how to retrieve the data from the API
-5. [Cursor field](../../cdk-python/incremental-stream.md) (Optional): Field to use as stream cursor. Can either be a string, or a list of strings if the cursor is a nested field.
-6. [Transformations](#transformations) (Optional): A set of transformations to be applied on the records read from the source before emitting them to the destination
-7. [Checkpoint interval](https://docs.airbyte.com/understanding-airbyte/airbyte-protocol/#state--checkpointing) (Optional): Defines the interval, in number of records, at which incremental syncs should be checkpointed
+The schema of a stream object is:
+
+```yaml
+Stream:
+  type: object
+  additionalProperties: false
+  required:
+    - name
+    - schema_loader
+    - retriever
+  properties:
+    "$options":
+      "$ref": "#/definitions/$options"
+    name:
+      type: string
+    primary_key:
+      "$ref": "#/definitions/PrimaryKey"
+    schema_loader:
+      "$ref": "#/definitions/SchemaLoader"
+    retriever:
+      "$ref": "#/definitions/Retriever"
+    stream_cursor_field:
+      type: object
+      oneOf:
+        - type: string
+        - type: array
+          items:
+            type: string
+    transformations:
+      type: array
+      items:
+        "$ref": "#/definitions/RecordTransformation"
+    checkpoint_interval:
+      type: integer
+```
+
+More details on streams and sources can be found in the [basic concepts section](../../cdk-python/basic-concepts.md).
 
 ### Data retriever
 
-The data retriever defines how to read the data for a Stream, and acts as an orchestrator for the data retrieval flow.
-There is currently only one implementation, the `SimpleRetriever`, which is defined by
+The data retriever defines how to read the data for a Stream and acts as an orchestrator for the data retrieval flow.
 
-1. [Requester](requester.md): Describes how to submit requests to the API source
-2. [Paginator](pagination.md): Describes how to navigate through the API's pages
-3. [Record selector](record-selector.md): Describes how to extract records from a HTTP response
-4. [Stream Slicer](stream-slicers.md): Describes how to partition the stream, enabling incremental syncs and checkpointing
+It is described by:
+
+1. [Requester](./requester.md): Describes how to submit requests to the API source
+2. [Paginator](./pagination.md): Describes how to navigate through the API's pages
+3. [Record selector](./record-selector.md): Describes how to extract records from a HTTP response
+4. [Stream Slicer](./stream-slicers.md): Describes how to partition the stream, enabling incremental syncs and checkpointing
```

Comment:
lowercase or the previous one uppercase? capitalization is a little inconsistent